eth/gasprice/locaFeeMarket/nativediscount.go

```diff
@@ -0,0 +1,318 @@
+package locaFeeMarket
+
+import (
+	"bytes"
+	"errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"math/big"
+)
+
+
+const (
+	historicalMeanGasPrice = 35783571428 //wei
+	historicalSDGasPrice = 849870638 //wei
+	maxDiscountedGasPrice  = historicalMeanGasPrice + historicalSDGasPrice/2 // =36208506747wei
+
+	coreTransferGasAmount = 21000 // hard-coded gas amount
+
+	functionSelectorSize = 4
+	transferDataLen = 4+32+32  // transfer(address,uint256): 4 bytes for func ID, 32 bytes for address (20 bytes padded to 32), 32 bytes for amount
```

Comment:
Technically this approach can be extrapolated to any contract address/function signature, correct? For example, we could make staking CORE on the `Staking.sol` contract a fee-reduced transaction type

Reply:
If it's possible we'll probably end up going in that direction for some key `core-genesis-contracts` actions. Which begs the question if we should already rearchitect the list of ERC20 token addresses +  list of function signatures into a 1-to-1 structure such that each fee-reduced contract is a struct containing:
- an address
- a list of valid function signatures, where each signature has a data length

The above struct is probably off by a bit, but you get the idea. Thoughts?

Reply:
> Technically this approach can be extrapolated to any contract address/function signature, correct? For example, we could make staking CORE on the `Staking.sol` contract a fee-reduced transaction type

correct and part of our design goals. we want this mechanism to be easily extendable.

Reply:
> If it's possible we'll probably end up going in that direction for some key `core-genesis-contracts` actions. Which begs the question if we should already rearchitect the list of ERC20 token addresses + list of function signatures into a 1-to-1 structure such that each fee-reduced contract is a struct containing:
> 
> * an address
> * a list of valid function signatures, where each signature has a data length
> 
> The above struct is probably off by a bit, but you get the idea. Thoughts?

see response above as to the erc20-compliant contract address. I would not add an additional 'discounted-functions' column due to KISS as well as having no immediate value